core/src/com/group66/game/cannon/GameManager.java

```diff
@@ -0,0 +1,332 @@
+     * @param batch the batch used to draw with
+     * @param delta the delta
+     */
+    public void draw(AbstractGameScreen gameScreen, SpriteBatch batch, float delta) {
+        int xoffset = Config.SINGLE_PLAYER_OFFSET;
+        if (isSplit) {
+            xoffset = Config.SEGMENT_OFFSET * segmentOffset;
+        }
+        
+        //draw the background
+        batch.draw(gameScreen.getBackground(), xoffset + Config.BORDER_SIZE_SIDES, Config.BORDER_SIZE_BOT,
+                Config.LEVEL_WIDTH, Config.LEVEL_HEIGHT);
+        
+        /* Draw the brick wall */
+        batch.draw(gameScreen.getBrickWall(), roofHitbox.x, roofHitbox.y + 10, roofHitbox.width, roofHitbox.height);
+        
+        /* Update the balls */
+        ballManager.update(batch, delta);
+        
+        //draw cannon
+        cannon.draw(batch);
+        
+        /* Draw the score */
+        textDrawer.draw(batch, "Score: " + scoreKeeper.getCurrentScore(), xoffset + Config.SCORE_OFFSET,
+                Config.SCORE_OFFSET);
+    }
+
+    /**
+     * Shift clone.
+     *
+     * @param other the other
+     */
+    public void shiftClone(GameManager other) {
+        //TODO change to iterator
+        for (Ball b : other.getBallManager().getBallsStaticManager().getBallStaticList()) {
+            float xpos = Config.SEGMENT_OFFSET * segmentOffset + b.getX();
+            ballManager.getBallsStaticManager().addStaticBall(b.getType(), xpos, b.getY());
+        }
+    }
+    
+    /**
+     * Ball check roof.
+     *
+     * @param ball the ball
+     */
+    public void ballCheckRoof(Ball ball) {
+        if (ball.getTopHitbox().overlaps(roofHitbox)) {
+            System.out.println("Attach ball to top");
+            ball.setSpeed(0);
+            BallSnap.snapBallToRoof(ball, roofHitbox.y + ROOF_OFFSET, this.isSplit, this.segmentOffset);
+            
+            ballManager.addBallDeadlist(ball);
+            ballManager.addStaticBallToBeAdded(ball);
+            //ballDeadList.add(ball);
+            //ballToBeAdded.add(ball);
+        }
+    }
+    
+    /**
+     * Move balls down.
+     */
+    public void moveBallsDown() {
+        if (getBallCount() >= Config.NBALLS_ROW_DOWN && canShoot()) {
+            System.out.println("Move balls down");
+            moveRowDown();
+            setBallCount(0);
+        }
+    }
+
+    /**
+     * Update balls, this includes the ball lists and the graph.
+     *
+     * @param delta the delta
+     */
+    public void update(float delta) {
+        /* Start counting time*/
+        timeKeeper.universalTimeCounter(delta);
+        
+        /* Check if a ball hasn't been shot in the past 10 seconds */
+        timeKeeper.didHeShoot();
+        
+        /* Check shooting balls */
+        // NB. Currently only 1 ball can be shot at a time in the game nevertheless the
+        // current BallList implementation is kept for versatility and to be future proof
+        // TODO change to iterator
```

Comment:
Eclipse can create a list of all TODOs in the code. It's in: Window -> Show View -> Tasks